sdk/core/core-amqp/src/ConnectionContextBase.ts

```diff
@@ -192,7 +196,16 @@ export module ConnectionContextBase {
           parameters.config.sharedAccessKeyName,
           parameters.config.sharedAccessKey
         ),
-      dataTransformer: parameters.dataTransformer || new DefaultDataTransformer()
+      dataTransformer: parameters.dataTransformer || new DefaultDataTransformer(),
+      refreshConnection() {
+        const connection = new Connection(connectionOptions);
+        this.wasConnectionCloseCalled = false;
+        this.connectionLock = `${Constants.establishConnection}-${generate_uuid()}`;
+        this.negotiateClaimLock = `${Constants.negotiateClaim} - ${generate_uuid()}`;
+        this.connection = connection;
+        this.connectionId = connection.id;
+        this.cbsSession = new CbsClient(connection, connectionLock);
```

Comment:
Will this use the connectionLock from line 184?

Reply:
I missed that!!!!!

Reply:
```suggestion
        this.cbsSession = new CbsClient(connection, this.connectionLock);
```

?

Reply:
Nice catch, @chradek !

This was not pointed out in the discussion there - https://github.com/Azure/azure-sdk-for-js/pull/8444#discussion_r412617970 and I just assumed connectionLock doesn't need its own variable.

Also, why does connection need its own variable? Any reason why I shouldn't reference this.connection?

Reply:
It shouldn't need its own variable. I think I just did that in track1 out of habit so I wouldn't have to worry about what order I assign things to `this`.